system_tests/system_tests.go

```diff
@@ -281,6 +288,64 @@ CONSOLE_LOG=new`
 		})
 	})
 
+	Context("PVC resize", func() {
+		var cluster  *rabbitmqv1beta1.RabbitmqCluster
+
+		AfterEach(func() {
+			Expect(rmqClusterClient.Delete(context.TODO(), cluster)).To(Succeed())
+		})
+
+		BeforeEach(func() {
+			// volume expansion is not supported in kinD which is use in github action
+			if os.Getenv("SUPPORT_VOLUME_EXPANSION") == "false" {
+				Skip("SUPPORT_VOLUME_EXPANSION is set to false; skipping volume expansion test")
+			}
+
+			cluster = newRabbitmqCluster(namespace, "resize-rabbit")
+			cluster.Spec.Persistence = rabbitmqv1beta1.RabbitmqClusterPersistenceSpec{
+				StorageClassName: pointer.StringPtr("persistent-test"),
+			}
+			Expect(createRabbitmqCluster(ctx, rmqClusterClient, cluster)).To(Succeed())
+			waitForRabbitmqRunning(cluster)
+		})
+
+		It("allows volume expansion", func() {
+			podUID := pod(ctx, clientSet, cluster, 0).UID
+			output, err := kubectlExec(namespace, statefulSetPodName(cluster, 0), "df", "/var/lib/rabbitmq/mnesia")
+			Expect(err).ToNot(HaveOccurred())
+			previousDiskSize, err := strconv.Atoi(strings.Fields(strings.Split(string(output), "\n")[1])[1])
+
+			storageClass := &storagev1.StorageClass{}
+			Expect(rmqClusterClient.Get(ctx, types.NamespacedName{Name: storageClassName, Namespace: namespace}, storageClass)).To(Succeed())
+			Expect(*storageClass.AllowVolumeExpansion).To(BeTrue(), fmt.Sprintf(" 'AllowVolumeExpansion' set to false for storage class %s", storageClassName))
```

Comment:
I think this check belongs in the `BeforeSuite`. If the Storage Class does not allow volume expansion, it's not the Operator's fault because the Operator is not responsible for setting up and configuring the Storage Class. I think we should fail earlier, not inside the test.

Reply:
removed